src/dodal/plans/motor_util_plans.py

```diff
@@ -0,0 +1,100 @@
+import uuid
+from typing import Any, Dict, Generator
+
+from bluesky import plan_stubs as bps
+from bluesky.preprocessors import finalize_wrapper, pchain
+from bluesky.utils import Msg, make_decorator
+from ophyd_async.core import Device
+from ophyd_async.epics.motion import Motor
+
+from dodal.common import MsgGenerator
+
+
+class MoveTooLarge(Exception):
+    def __init__(
+        self, axis: Motor, maximum_move: float, position: float, *args: object
+    ) -> None:
+        self.axis = axis
+        self.maximum_move = maximum_move
+        self.position = position
+        super().__init__(*args)
+
+
+def _check_and_cache_values(
+    device: Device,
+    smallest_move: float,
+    maximum_move: float,
+    home_position: float = 0,
+) -> Generator[Msg, Any, Dict[Motor, float]]:
+    """Caches the positions of all Motors on specified device if they are within
+    smallest_move of home_position. Throws MoveTooLarge if they are outside maximum_move
+    of the home_position
+    """
+    positions = {}
+    motors_to_move = [axis for _, axis in device.children() if isinstance(axis, Motor)]
+    for axis in motors_to_move:
+        position = yield from bps.rd(axis)
+        if abs(position - home_position) > maximum_move:
+            raise MoveTooLarge(axis, maximum_move, position)
+        if abs(position - home_position) > smallest_move:
+            positions[axis] = position
+    return positions
+
+
+def home_and_reset_wrapper(
+    plan: MsgGenerator,
+    device: Device,
+    home_position: float,
```

Comment:
are we likely to ever have this not be 0, and if it is ever not 0 isn't it then likely to be different for every motor?

how would you feel about this instead taking a list of motors, and a list of homes or default to 0 for everything?